### PR TITLE
[dns2] Add support for IPv6 UDP DNS servers

### PR DESCRIPTION
**Pull request by karusher**
To instantiate a UDP IPv6 DNS server, it is necessary to pass an object with type set as 'udp6' as a constructor argument. This will cause dns2 to open the right kind of socket.

See https://github.com/song940/node-dns/blob/master/server/udp.js

This commit updates the UDP objects to add this facility.

The address has been made optional in the UDP listen signature to allow the socket to choose the right all interface address.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/song940/node-dns/blob/master/server/udp.js
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## dns2 ([<kbd>unpkg</kbd>](https://unpkg.com/browse/dns2@latest/))
was missing the following properties:
1. DOHServer
1. createDOHServer
1. TCPClient
1. DOHClient
1. UDPClient

<details>
<summary>as well as these 9 other properties...</summary>
<p>GoogleClient, once, on, listenerCount, errorMonitor, captureRejectionSymbol, captureRejections, defaultMaxListeners, EventEmitter</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e2a7e0b076b54e2d2b5897049e7f4fdd05a60364
</p>

File: types/dns2/index.d.ts [hunk 1]
```diff
@@ -85,6 +85,10 @@ declare namespace DNS {
         data?: string;
     }
 
+    interface UdpDnsServerOptions {
+        type: string;
```
Comment (pimterry) [inline types/dns2/index.d.ts:89]: I think we could tighten this up slightly:

```suggestion
        type: 'udp4' | 'udp6';
```

Looks like the type gets passed up to Node.js's APIs, which do explicitly limit to just those values.

**Review by pimterry — COMMENTED** [review]
One small tweak, otherwise looks good to me

**Comment by karusher** [comment]
Made the change! Thank you for the review.

**Review by pimterry — APPROVED** [review]

**Comment by karusher** [comment]
Ready to merge

**Pull request merged**